src/client/websocket/WebSocketShard.js

```diff
@@ -475,12 +475,22 @@ class WebSocketShard extends EventEmitter {
       return;
     }
     const hasGuildsIntent = new Intents(this.manager.client.options.intents).has(Intents.FLAGS.GUILDS);
-    // Step 2. Create a 15s timeout that will mark the shard as ready if there are still unavailable guilds
+    // Step 2. Create a timeout that will mark the shard as ready if there are still unavailable guilds
+    // * the timeout should be 15 seconds by default, but this can be optionally be changed in the client options
+
+    var guildTimeoutAmount = 15000;
+
+    if (this.manager.client.options.fetchGuildTimeout) {
+      guildTimeoutAmount = this.manager.client.options.fetchGuildTimeout;
+    }
+
+    var guildTimeoutAmountInSeconds = guildTimeoutAmount / 1000;
```

Comment:
🧹
```suggestion
    const guildTimeoutAmount = this.manager.client.options.fetchGuildTimeout || 1500;
```

Reply:
You sure you want 0 to be treated as default? If not use ??

Reply:
what about setting the default here
https://github.com/discordjs/discord.js/blob/d16ada97087e0324134484c9c0724b52a2839bdf/src/util/Options.js#L110-L146

Reply:
I mean yes that would be preferred once validation is added 😎

Reply:
> You sure you want 0 to be treated as default? If not use ??

If `15000` is added as a default option this isn't needed as `this.manager.client.options.fetchGuildTimeout` can be used... However, why you should use 0? Does this mean no timeout at all or no guilds at all?